app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsActivity.java

```diff
@@ -4,23 +4,55 @@
 import android.content.Intent;
 import android.databinding.DataBindingUtil;
 import android.os.Bundle;
-import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
 import android.support.v7.app.ActionBar;
+import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
 import android.view.MenuItem;
 
 import com.invisibleteam.goinvisible.R;
 import com.invisibleteam.goinvisible.databinding.SettingsActivityBinding;
 import com.invisibleteam.goinvisible.model.ClearingInterval;
+import com.invisibleteam.goinvisible.mvvm.common.StringTypesAdapter;
 import com.invisibleteam.goinvisible.util.SharedPreferencesUtil;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import static com.invisibleteam.goinvisible.mvvm.settings.SettingsViewModel.SettingsViewModelCallback;
+
 public class SettingsActivity extends AppCompatActivity {
 
     public static Intent buildIntent(Context context) {
         return new Intent(context, SettingsActivity.class);
     }
 
+    private final SettingsViewModelCallback viewModelCallback = new SettingsViewModelCallback() {
+
+        @Override
+        public void onOpenIntervalsDialog() {
+            openIntervalsDialog();
+        }
+
+        @Override
+        public void onClearCache() {
+            viewModel.setIntervalName(ClearingInterval.DAY.getIntervalFormattedName(SettingsActivity.this));
```

Comment:
Do we need to clear cache and set interval to default?
I think that leaving it as it is and just stopping the "clearing service" would be enough.